### PR TITLE
Support branches with slashes in `comments-time-machine-links`

### LINKED ISSUE #2901: `comments-time-machine-links` doesn't work with branches that have slashes
Example URL:
https://github.com/Cog-Creators/Red-DiscordBot/issues/2988

### PR DESCRIPTION
**Pull request by yakov116**
1. LINKED ISSUES:
   Fixes #2901

2. TEST URLS:
   ```
	https://github.com/Cog-Creators/Red-DiscordBot/blob/develop/cogs/downloader.py?rgh-link-date=2018-03-19T21%3A24%3A31Z
`

3. SCREENSHOT:
   None

Gatcha:
```
	https://github.com/Cog-Creators/Red-DiscordBot/tree/V3/develop@%7B2019-09-07T19:14:39Z%7D?rgh-link-date=2020-06-02T18%3A02%3A49Z
```

Github URL does not drop the `@%7B2019-09-07T19:14:39Z%7D` from the branch name

### CODE REVIEW

File: source/features/comments-time-machine-links.tsx [hunk 1]
```diff
@@ -78,7 +99,7 @@ async function showTimemachineBar(): Promise<void | false> {
 		}
 
 		const parsedUrl = new GitHubURL(location.href);
-		parsedUrl.branch = `${parsedUrl.branch}@{${date}}`;
+		parsedUrl.branch = await getDatedSha(parsedUrl, date);
```
Comment (fregante) [inline source/features/comments-time-machine-links.tsx:102]: Can you set the old href first and then update it when the API returns? This way the bar appears faster (and still works most of the time anyway)

Comment (yakov116) [reply source/features/comments-time-machine-links.tsx:102]: Yes

Comment (yakov116) [reply source/features/comments-time-machine-links.tsx]: Done

**Comment by yakov116** [comment]
@fregante off-topic:

I figured this out when I attempted https://github.com/sindresorhus/refined-github/issues/2927#issuecomment-640151498. Feels good to know my 2 hours of trying came to use ☕

File: source/features/comments-time-machine-links.tsx [hunk 2]
```diff
@@ -5,9 +5,31 @@ import elementReady from 'element-ready';
...
+import {getRepoURL, isPermalink, getRepoGQL} from '../github-helpers';
+
+const updateURLtoDatedSha = async (url: GitHubURL, date: string) => {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			ref(qualifiedName: "${url.branch}") {
+				target {
+					... on Commit {
+						history(first: 1, until: "${date}") {
+							nodes{
```
Comment (fregante) [inline source/features/comments-time-machine-links.tsx:20]:
```suggestion
							nodes {
```

**Comment by khoiclub** [comment]
# 3406

**Comment by fregante** [comment]
This PR is supposed to fix bugs related to "branches with slashes" but it doesn't include a testing URL with "branches with slashes"

**Comment by fregante** [comment]
Here's a link: 

```
Raw: https://github.com/Cog-Creators/Red-DiscordBot/blob/V3/develop/.gitignore

Pre-baked: https://github.com/Cog-Creators/Red-DiscordBot/blob/V3/develop/.gitignore?rgh-link-date=2018-03-19T21%3A24%3A31Z
```

**Comment by yakov116** [comment]
@fregante sorry that was a mistake. Will be more careful next time.

**Pull request merged**